src/savings-pool-client.ts

```diff
@@ -62,8 +58,77 @@ export default class SavingsPoolClient {
    * @returns The operation hash
    */
   public async redeem(lpTokenAmount: BigNumber): Promise<TransactionOperation | TransactionWalletOperation> {
-    const liquidityPoolContract = await this.tezos.wallet.at(this.savingsPoolAddress)
+    const savingsPoolContract = await this.tezos.wallet.at(this.savingsPoolAddress)
     const sendArgs = { amount: 0, mutez: true }
-    return await liquidityPoolContract.methods['redeem'](lpTokenAmount).send(sendArgs)
+    return await savingsPoolContract.methods['redeem'](lpTokenAmount).send(sendArgs)
+  }
+
+  /**
+   * Get the interest rate of the pool.
+   */
+  public async getInterestRateAPY(): Promise<BigNumber> {
+    const savingsPoolContract = await this.tezos.wallet.at(this.savingsPoolAddress)
+    const savingsPoolStorage: any = await savingsPoolContract.storage()
```

Comment:
As a general rule I'm *always* already resolving storage and a handle to these contracts in the frontend. 

Can you make them optional parameters that'll resolve lazily if they're not passed in?